odoo/models.py

```diff
@@ -4445,11 +4451,89 @@ def unlink(self):
             ))
             ir_attachment_unlink |= Attachment.browse(row[0] for row in cr.fetchall())
 
+            # delete fallback value in ir.default for company dependent fields
+            if (fields_ := self.env.registry.many2one_company_dependents[self._name]):
+                IrModelFields = self.env["ir.model.fields"]
+                field_ids = tuple(IrModelFields._get_ids(field.model_name).get(field.name) for field in fields_)
+                sub_ids_json_text = tuple(str(id_) for id_ in sub_ids)
+
+                if self._context.get(MODULE_UNINSTALL_FLAG):
+                    self.env.execute_query(SQL(
+                        """
+                        DELETE FROM ir_default
+                        WHERE field_id IN %(field_ids)s
+                        AND json_value IN %(ids)s
+                        """,
+                        field_ids=field_ids,
+                        ids=sub_ids_json_text,
+                    ))
+                elif res := self.env.execute_query(SQL(
+                    """
+                    SELECT json_value::int4, field_id
+                    FROM ir_default
+                    WHERE field_id IN %(field_ids)s
+                    AND json_value IN %(ids)s
+                    LIMIT 1
+                    """,
```

Comment:
flush is not necessary since `self.env.flush_all()` has been done before in the `unlink`

> Why not using the ORM here

wrote too many SQL, and forgot we have powful ORM 🤣